zipline/finance/performance/tracker.py

```diff
@@ -523,12 +517,31 @@ def _calculate_rolling_expected_shortfall(self, data_portal):
         )
 
         asset_returns = zp.asset_returns_for_cvar(
-            assets=assets,
+            assets=list(weights.columns),
             benchmark=self.benchmark_asset,
             data_portal=data_portal,
             end_date=sim_params.end_session,
             lookback_days=len(sim_params.sessions) + days_before_start,
         )
+        asset_returns_values = asset_returns.values
+
+        def rolling_shortfall():
+            cvar_cutoff = zp.DEFAULT_CVAR_CUTOFF
+            out = np.full(len(weights), np.nan)
+
+            # Compute from back to front, since that simplifies the task of
+            # aligning the correct row of `weights with the correct slice of
+            # `asset_returns`.
+            last_end = -len(weights)
+            end = -1
+            while end >= last_end:
+                # TODO: Bail if we don't have enough input data.
```

Comment:
I don't think we want to bail, we want to just compress the window and use what we have, down to a certain point.